zigup.zig

```diff
@@ -421,19 +420,48 @@ fn listCompilers(allocator: *Allocator) !void {
     }
 }
 
-fn printDefaultCompiler(allocator: *Allocator) !void {
+fn cleanCompilers(allocator: *Allocator, version_name: [:0]u8) !void {
```

Comment:
So, I'm not sure how useful it is to implementing cleaning a specific compiler version.  The user can do that easy enough by just deleting the directory themselves.  What's more useful is the ability to all unused compilers.  So this goes hand-in-hand with implementing the "keep" functionality.  My idea was to include an empty file, probably named "keep" in each compiler install directory if the user wants to keep it.  So the clean command would just clean all compilers that don't have a "keep" marker file.

Reply:
Ok. I see now. Ill try to impliment your idea with keep. If a user wants to keep a certain master, should they just give it with the hash, like `zigup keep 0.6.0+123456`? Thank you for all this feedback, I am becoming better. I have also removed the unnessary elses.

Reply:
>  If a user wants to keep a certain master, should they just give it with the hash, like zigup keep 0.6.0+123456?

Basically yeah, if they want to keep any version it's just `zigup keep VERSION`.  Maybe it would also work with `master`?  Not sure on that one, since the user may not have master downloaded....